src/components/Images.js

```diff
@@ -0,0 +1,196 @@
+import img6 from "../img/6.png";
+
+const defaultImg = 'https://via.placeholder.com/261x261?text=CLICK'
+
+function Images() {
+    const imgs = [
+        {
+            name: 1,
+            loc: img1,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 2,
+            loc: img2,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 3,
+            loc: img3,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 4,
+            loc: img4,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 5,
+            loc: img5,
+            flipped: true,
+            completed: true
+        },
+        {
+            name: 6,
+            loc: img6,
+            flipped: true,
+            completed: true
+        }
+    ]
+
+    const [finalGame, setfinalGame] = useState([])
+    const [preName, setPreName] = useState()
+    const [points, setPoints] = useState(0)
+    const [selectedImg, setSelectedImg] = useState()
+    const [hideStart, setHideStart] = useState()
+    const [playerName, setPlayerName] = useState()
+    const [timer, setTimer] = useState(3)
+    
+    useEffect(() => {
+        // const finalImages = [...imgs, ...imgs]
+        let finalImages = []
+        imgs.map(img => {
+            finalImages.push(JSON.parse(JSON.stringify(img)))
+            finalImages.push(JSON.parse(JSON.stringify(img)))
+        })
+        //2x Shuffling the images to make it more random
+        let shuffle = finalImages.sort(() => Math.random() - 0.5).map((img) => img)
+        let finalBoard = shuffle.sort(() => Math.random() - 0.5).map((img) => img)
+        setfinalGame(finalBoard)
+    },[])
+
+    const startGame = () => {
+        // test()
+        let stop = 0        
+        const timerClock = setInterval(() => {
+            setTimer(preTimer => preTimer - 1)
+            stop++
+            setHideStart(1)
+            if(stop === 3) {
+                clearInterval(timerClock)
+                stop = 0
+                setHideStart()
+            }
+        }, 1000);
+
+        setTimeout(() => {
+            const hideImg = finalGame.map(img => {
+                if(img) {
+                    img.flipped = !img.flipped
+                    img.completed = !img.completed
+                }
+                return {...img}
+            })
+            console.log(hideImg)
+            setfinalGame(hideImg)
+            
+        }, 3000)
+    }
+
+    const cleanSelectedImg = (previous, current) => {
+        setTimeout(() => {
+            setfinalGame(finalGame.map((img, index) => {
+                if(index === previous || index === current) {
+                    // console.log(img.name)
+                    img.flipped = !img.flipped
+                }
+                  return {...img}
+              }))
+        }, 400)
```

Comment:
Why so many random timeouts?

Reply:
I have used 3 timeouts
1) hideImg - this does not allow user to flip images until game is not started
2) timerClock - game start timer 
3) cleanSelectedImg - if two different images are selected this will hide them again

Reply:
I mean the numbers Its 400, 3000, 1000
Use constant to define timeouts for certain operation 
Like: `START_GAME_TIMEOUT = 3000`